roomserver/storage/postgres/membership_table.go

```diff
@@ -124,6 +124,15 @@ var selectKnownUsersSQL = "" +
 	"  SELECT DISTINCT room_nid FROM roomserver_membership WHERE target_nid=$1 AND membership_nid = " + fmt.Sprintf("%d", tables.MembershipStateJoin) +
 	") AND membership_nid = " + fmt.Sprintf("%d", tables.MembershipStateJoin) + " AND event_state_key LIKE $2 LIMIT $3"
 
+// selectLocalServerInRoomSQL is an optimised case for checking if we, the local server,
+// are in the room by using the target_local column of the membership table. Normally when
+// we want to know if a server is in a room, we have to unmarshal the entire room state which
+// is expensive. The presence of a single row from this query suggests we're still in the
+// room, no rows returned suggests we aren't.
+const selectLocalServerInRoomSQL = "" +
+	"SELECT room_nid FROM roomserver_membership WHERE target_local = true AND membership_nid = 3 AND room_nid = $1 " +
+	"FETCH FIRST 1 ROWS ONLY"
```

Comment:
Why not `LIMIT 1`?

Reply:
I had in mind that there was an optimisation around `FETCH FIRST 1 ROWS ONLY` vs `LIMIT 1`, but looking into it I don't think it applies in PostgreSQL. Switched it to `LIMIT 1`.